src/classes/storage.ts

```diff
@@ -112,13 +139,24 @@ export class Storage {
   /**
    * @description Update StopGroup or StopCluster with new informations like name, latitude, longitude, etc.
    * @param {Gpuid} placeGpuid - Ground place unique identifier of the place to update.
-   * @param {UpdateStopProperties} propertiesToUpdate - Properties to update {name, lattitude, longitude}.
+   * @param {UpdateGroundPlaceProperties} propertiesToUpdate - Properties to update {name, lattitude, longitude}.
    * @param {GroundPlaceType} placeType - The type of the place to update, can be StopGroup or StopCluster.
    * @returns {void}
    */
-  public updatePlace(placeGpuid: Gpuid, propertiesToUpdate: UpdateStopProperties, placeType: GroundPlaceType): void {
+  public updatePlace(
+    placeGpuid: Gpuid,
+    propertiesToUpdate: UpdateGroundPlaceProperties,
+    placeType: GroundPlaceType,
+  ): void {
     const groundPlace: GroundPlace = this.getPlaceByGpuid(placeGpuid, placeType);
 
+    // Filter the `propertiesToUpdate` object with values properly defined
+    Object.keys(propertiesToUpdate).forEach((key: string) => {
+      if (propertiesToUpdate[key] === undefined) {
```

Comment:
if which case do you have undefined properties ?

Reply:
From the switch-statement inside the `applyGroundPlacesDiff` method:

```js
case ActionType.UPDATE_STOP_GROUP:
    return this.updateStopGroup(groundPlaceGpuid, {
      latitude: params.latitude,
      longitude: params.longitude,
      name: params.name,
    });
```
Here, I'll pass to `updateStopGroup` all params that can be update. But inside the GroundPlacesDiff file, maybe you only want to update the name or something else. In this case, the other params are pass with `undefined`values.
So the purpose of this `Object.keys` is to sanitize the parameters pass to this method, and only apply the params that are well defined

Reply:
ok got it, thanks